### PR TITLE
fix: update proxy data type for response handler input

### PR DESCRIPTION
**Pull request by utsabc**
## What are the changes introduced in this PR?

This PR attempts to clean up type definition inconsistencies for the Proxy Routes for both V0 and V1. The main objective is make sure the type definitions are on par with server type definitions.

Secondly the metadata which is part of the proxy request params for both V0 and V1 Proxy route was incorrectly set to destination response, which is also cleaned as part of this PR

`DeliveryV0Response` -> Response from transformer to server for v0 Proxy API 
[Server struct ref](https://github.com/rudderlabs/rudder-server/blob/e1789a5c7cc070bf9d5a7d8ba0e3c7e67886bb5d/router/transformer/transformer_proxy_adapter.go#L31)
[Usage](https://github.com/rudderlabs/rudder-server/blob/e1789a5c7cc070bf9d5a7d8ba0e3c7e67886bb5d/router/transformer/transformer_proxy_adapter.go#L88)

`DeliveryV1Response` -> Response from transformer to server v1 Proxy API
[Server struct ref](https://github.com/rudderlabs/rudder-server/blob/e1789a5c7cc070bf9d5a7d8ba0e3c7e67886bb5d/router/transformer/transformer_proxy_adapter.go#L37)
[Usage](https://github.com/rudderlabs/rudder-server/blob/e1789a5c7cc070bf9d5a7d8ba0e3c7e67886bb5d/router/transformer/transformer_proxy_adapter.go#L132)

`ProxyV0Request` -> Request from Server to transformer for v0 Proxy API
[Server struct ref](https://github.com/rudderlabs/rudder-server/blob/e1789a5c7cc070bf9d5a7d8ba0e3c7e67886bb5d/router/transformer/transformer_proxy_adapter.go#L26)
[Usage](https://github.com/rudderlabs/rudder-server/blob/e1789a5c7cc070bf9d5a7d8ba0e3c7e67886bb5d/router/transformer/transformer_proxy_adapter.go#L65)

`ProxyV1Request` -> Request from Server to transformer for v1 Proxy API
[Server struct ref](https://github.com/rudderlabs/rudder-server/blob/e1789a5c7cc070bf9d5a7d8ba0e3c7e67886bb5d/router/transformer/transformer.go#L65)
[Usage](https://github.com/rudderlabs/rudder-server/blob/e1789a5c7cc070bf9d5a7d8ba0e3c7e67886bb5d/router/transformer/transformer_proxy_adapter.go#L114)

`ProxyMetdata` -> Metadata used in proxy Payloads
[Server struct](https://github.com/rudderlabs/rudder-server/blob/e1789a5c7cc070bf9d5a7d8ba0e3c7e67886bb5d/router/transformer/transformer.go#L53)
[Usage](https://github.com/rudderlabs/rudder-server/blob/e1789a5c7cc070bf9d5a7d8ba0e3c7e67886bb5d/router/transformer/transformer.go#L67)
[Usage](https://github.com/rudderlabs/rudder-server/blob/e1789a5c7cc070bf9d5a7d8ba0e3c7e67886bb5d/router/transformer/transformer_proxy_adapter.go#L28)

## What is the related Linear task?

INT-1423

### Developer checklist

- [x] My code follows the style guidelines of this project

- [x] **No breaking changes are being introduced.**

- [x] All related docs linked with the PR?

- [ ] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [ ] Is the PR limited to 10 file changes?

- [x] Is the PR limited to one linear task?

- [x] Are relevant unit and component test-cases added?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

## Summary by CodeRabbit

- **New Features**
  - Enhanced delivery response handling with version-specific types for improved clarity and consistency.

- **Refactor**
  - Standardized response handling across various network handlers to accept parameter objects for better code maintainability.
  - Updated service methods to return new version-specific response types.

- **Bug Fixes**
  - Addressed issues with delivery failure event handling for certain destination services.

- **Documentation**
  - Updated internal documentation to reflect changes in response types and method signatures.

- **Tests**
  - Modified test data to align with the new response type structures and metadata fields.

### CODE REVIEW

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3030/test-report.html

**Comment by utsabc** [comment]
@coderabbitai review

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3030/test-report.html

File: src/services/destination/nativeIntegration.ts [hunk 1]
```diff
@@ -191,24 +191,22 @@ export class NativeIntegrationDestinationService implements DestinationService {
...
       }
-
-      let responseProxy = networkHandler.responseHandler(
-        {
-          ...processedProxyResponse,
-          rudderJobMetadata,
-        },
-        destinationType,
-      );
+      const responseParams = {
```
Comment (utsabc) [inline src/services/destination/nativeIntegration.ts:200]: This is the core part where the change has been done which affects all network handlers

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3030/test-report.html

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3030/test-report.html

File: src/v0/destinations/marketo/networkHandler.js [hunk 2]
```diff
@@ -3,9 +3,10 @@ const { marketoResponseHandler } = require('./util');
 const { proxyRequest, prepareProxyRequest } = require('../../../adapters/network');
 const { processAxiosResponse } = require('../../../adapters/utils/networkUtils');
 
-const responseHandler = (destinationResponse, destType) => {
+const responseHandler = (responseParams) => {
+  const { destinationResponse, destType,rudderJobMetadata } = responseParams;
```
Comment (chandumlg) [inline src/v0/destinations/marketo/networkHandler.js:7]:
```suggestion
  const { destinationResponse, destType, rudderJobMetadata } = responseParams;
```

File: src/v0/destinations/fb/networkHandler.js [hunk 3]
```diff
@@ -2,7 +2,8 @@ const { processAxiosResponse } = require('../../../adapters/utils/networkUtils')
 const { errorResponseHandler } = require('../facebook_pixel/networkHandler');
 const { prepareProxyRequest, proxyRequest } = require('../../../adapters/network');
 
-const destResponseHandler = (destinationResponse) => {
+const destResponseHandler = (responseParams) => {
```
Comment (chandumlg) [inline src/v0/destinations/fb/networkHandler.js:5]: Lets rename `destResponseHandler` to `responseHandler`?

File: src/v0/util/facebookUtils/networkHandler.js [hunk 4]
```diff
@@ -275,7 +275,8 @@ const errorResponseHandler = (destResponse) => {
   );
 };
 
-const destResponseHandler = (destinationResponse) => {
+const destResponseHandler = (responseParams) => {
```
Comment (chandumlg) [inline src/v0/util/facebookUtils/networkHandler.js:278]: Lets rename `destResponseHandler` to `responseHandler`?

**Review by chandumlg — APPROVED** [review]

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3030/test-report.html

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3030/test-report.html

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3030/test-report.html

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3030/test-report.html

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3030/test-report.html

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3030/test-report.html

**Comment by koladilip** [comment]
@coderabbitai review

**Pull request merged**